authenticator/src/main/java/com/beyondidentity/authenticator/sdk/AuthView.kt

```diff
@@ -40,23 +40,68 @@ class AuthView : LinearLayout {
     /**
      * Initialize the [AuthView]
      *
-     * @param loginUrl The http endpoint your API backend provides to start the OAUTH2 flow
-     * @param redirectUrl Where should the server redirect after successful authentication with the access code
+     * Use this initialization flow when your backend already has the Oauth configuration setup,
+     * client_id, client_secret, scope, response_type, etc...
+     *
+     * You can expect to get the access token in the Activity specified in the intent filter
+     *
+     * @param authorizationEndpoint The https authorization endpoint your API backend provides.
+     * @param redirectUri Where should the server redirect after authorization completes.
      * @param signupButtonListener OnClickListener for the Sign Up Button
      */
     fun initAuthView(
-        loginUrl: String,
-        redirectUrl: String,
+        authorizationEndpoint: String,
+        redirectUri: String,
         signupButtonListener: OnClickListener
     ) {
-        loginButton.setOnClickListener {
-            val url = "$loginUrl?redirect=$redirectUrl"
-            val builder = CustomTabsIntent.Builder()
+        val uri = Uri.parse(authorizationEndpoint).buildUpon().apply {
+            appendQueryParameter("redirect", redirectUri)
+        }.build()
+
+        loginButton.setOnClickListener { launchCustomTab(context, uri) }
+        signupButton.setOnClickListener(signupButtonListener)
+    }
 
-            val customTabsIntent = builder.build()
-            customTabsIntent.launchUrl(context, Uri.parse(url))
-        }
+    /**
+     * Initialize the [AuthView]
+     *
+     * Use this initialization flow to initiate and configure the oauth2 flow from your app to get
+     * the authorization_code. You can then use the authorization_code to get access token from your
+     * backend who will exchange it using the client_secret.
+     *
+     * @param clientId unique ID for a tenant's registered OIDC client.
+     * @param authorizationEndpoint The https authorization endpoint your API backend provides.
+     * @param redirectUri Where should the server redirect after authorization completes. Must match configured values.
+     * @param responseType either `code` or `id_token`
```

Comment:
We aren't going to support the implicit flow so I would remove `id_token`.

Reply:
@ismailmustafa technically "token" is the implicit flow haha If you have the course see Section 9 Hybrid OpenID Connect Flows.
![IMG_3FAC27AB36A7-1](https://user-images.githubusercontent.com/11654201/112203345-d361af00-8be8-11eb-9b2a-565493e9d436.jpeg)

Reply:
Ford mentioned that we don't support hybrid and I'm assuming devs won't care about only `id_token`
But yeah, what Anna said, implicit flow is basically when you get the access token without the authorization code step.

Reply:
Yea just making the distinction because we might want to support `code id_token` in the future.

Reply:
Got it, I stand corrected. Leaving this here for reference: 
![image](https://user-images.githubusercontent.com/6578679/112212461-67387880-8bf3-11eb-86c1-35cee25b3607.png)